### PR TITLE
Feature/register

### PR DESCRIPTION
**Pull request by joeythegod**
Create following pages:
-Login
-Register
-main
-event

Integrate following api:
-user login (POST)
-user register (POST)
-event create (POST)
-event fetch (GET)

### CODE REVIEW

File: lib/fetch.dart [hunk 1]
```diff
@@ -1,12 +1,90 @@
...
+      'password': password,
+      'email': email,
+    },
+  );
+  if (response.statusCode >= 200 && response.statusCode <= 210) {
+    return User.fromJson(json.decode(response.body));
+  }
+  else {
+    throw Exception('Failed to create user.');
+  }
```
Comment (0charleschen0) [inline lib/fetch.dart]: Besides throw exception, we should consider which error causes users cannot register.

File: lib/fetch.dart [hunk 2]
```diff
@@ -1,12 +1,90 @@
...
+
+  factory User.fromJson(Map<String, dynamic> json) {
+    return User(
+      username: json['username'],
+      password: json['password'],
+      email: json['email'],
+      userId: json['userId'],
+    );
+  }
+}
```
Comment (0charleschen0) [inline lib/fetch.dart]: Better to extract it to another class.

File: lib/fetch.dart [hunk 3]
```diff
@@ -1,12 +1,90 @@
...
+  factory Event.fromJson(Map<String, dynamic> json) {
+    return Event(
+      userId: json['userId'],
+      title: json['title'],
+      startTime: json['startTime'],
+      endTime: json['endTime'],
+      content: json['content'],
+    );
+  }
+}
```
Comment (0charleschen0) [inline lib/fetch.dart]: Better to extract it to another class.

File: lib/fetch.dart (hunk 1 above)
Comment (0charleschen0) [reply lib/fetch.dart]: Same as other functions

File: lib/pages/home.dart [hunk 4]
```diff
@@ -27,24 +34,118 @@ class MainPage extends StatelessWidget {
...
+                  ),
+                  ListTile(
+                    leading: Icon(Icons.send),
+                    title: Text('Create!'),
+                    onTap: () async {
+                      await showDialog(
+                        context: context,
+                        barrierDismissible: false,
+                        builder: (context) => AlertDialog(
+                          content: _createEvent(_user.userId, _controller_title.text, _controller_startTime.text, _controller_endTime.text, _controller_content.text),
```
Comment (0charleschen0) [inline lib/pages/home.dart]: Please separate this to multiple lines.

**Review by 0charleschen0 — APPROVED** [review]
Please merge this PR and address my comments and the following in next **PR**

1. Unify the naming rule: I see some naming are camelCase 
2. Extract methods out for looooooong method ex: home.dart#build
3. Add comments/document to make your code easier to understand

**Comment by joeythegod** [comment]
> Please merge this PR and address my comments and the following in next **PR**
> 
> 
> 
> 1. Unify the naming rule: I see some naming are camelCase 
> 
> 2. Extract methods out for looooooong method ex: home.dart#build
> 
> 3. Add comments/document to make your code easier to understand

Ok! I will fix the above comment and push a new commit! Thanks!

**Pull request merged**